test/test_call.py

```diff
@@ -0,0 +1,93 @@
+from .support import HPyTest
+
+
+class TestCall(HPyTest):
+    def test_hpy_call(self):
+        import pytest
+        mod = self.make_module("""
+            HPyDef_METH(call, "call", call_impl, HPyFunc_VARARGS)
+            static HPy call_impl(HPyContext ctx, HPy self,
+                              HPy *args, HPy_ssize_t nargs)
+            {
+
+                HPy f, f_args;
+                HPy f_kw = HPy_NULL;
+                if (!HPyArg_Parse(ctx, NULL, args, nargs, "OO|O", &f, &f_args,
+                                  &f_kw))
+                    return HPy_NULL;
+                return HPy_Call(ctx, f, f_args, f_kw);
+            }
+            @EXPORT(call)
+            @INIT
+        """)
+
+        def f(a, b):
+            return a + b
+
+        assert mod.call(f, (1, 2)) == 3
```

Comment:
maybe we should also test what happens if you pass something which is not a tuple and/or a dict?

Reply:
Good idea. I'll add some.

Reply:
The answer is that C Python segfaults in some cases and works fine in others (e.g. a list of args). I've added tests and changed the HPy implementation to check the types of handles rather than segfault.

Reply:
I wonder if doing these extra check costs any performance. Do you feel like adding microbenchmarks?
If they are costly, we could do the check only in debug mode.

Reply:
I'm guessing not a whole lot, but I'll add the microbenchmarks.

Reply:
Microbenchmarks added.